From b8275ba2c9f2cec20930728929d125b1af503c58 Mon Sep 17 00:00:00 2001
From: Pedro Palazon Candel <pedro@joyent.com>
Date: Fri, 15 Jun 2018 19:47:04 +0200
Subject: [PATCH] joyent/node-cueball#138 want method to know if Agent is
 stopped Reviewed by: Alex Wilson <alex.wilson@joyent.com> Approved by: Alex
 Wilson <alex.wilson@joyent.com>

---
 CHANGES.adoc       | 8 ++++++++
 docs/api.adoc      | 5 +++++
 lib/agent.js       | 4 ++++
 package.json       | 2 +-
 test/agent.test.js | 3 +++
 5 files changed, 21 insertions(+), 1 deletion(-)

diff --git a/CHANGES.adoc b/CHANGES.adoc
index ae2b08c..9a90ea8 100644
--- a/CHANGES.adoc
+++ b/CHANGES.adoc
@@ -6,6 +6,14 @@ toc::[]
 
 ## v2.x
 
+### v2.5.3
+
+Maintenance release.
+
+Bugs fixed:
+
+ - #138 added `isStopped()` method to `Agent`.
+
 ### v2.5.2
 
 Maintenance release.
diff --git a/docs/api.adoc b/docs/api.adoc
index 968ca8b..ae99c37 100644
--- a/docs/api.adoc
+++ b/docs/api.adoc
@@ -136,6 +136,11 @@ held by the pool). The `stop()` method should be used on an Agent if your
 program does not plan to use it again and wishes to be able to exit node
 without calling `process.exit()`.
 
+### `#isStopped()`
+
+Boolean, is this Agent stopped?. Useful to prevent exceptions being thrown
+when calling `stop()` more than once.
+
 ### `#getPool([host])`
 
 Returns the connection pool this agent has created for the input host. If no
diff --git a/lib/agent.js b/lib/agent.js
index fbae298..6b3e292 100644
--- a/lib/agent.js
+++ b/lib/agent.js
@@ -491,6 +491,10 @@ CueBallAgent.prototype.createConnection = function (options, connectListener) {
 	throw (new Error('UNIX domain sockets not supported'));
 };
 
+CueBallAgent.prototype.isStopped = function () {
+	return (this.cba_stopped);
+};
+
 function CueBallHttpAgent(options) {
 	options.protocol = 'http';
 	if (options.defaultPort === undefined)
diff --git a/package.json b/package.json
index 39d78f9..321221f 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "cueball",
-  "version": "2.5.2",
+  "version": "2.5.3",
   "description": "manage a pool of connections to a multi-node service where nodes are listed in DNS",
   "main": "lib/index.js",
   "dependencies": {
diff --git a/test/agent.test.js b/test/agent.test.js
index 2779f3f..38d0153 100644
--- a/test/agent.test.js
+++ b/test/agent.test.js
@@ -73,8 +73,11 @@ mod_tape.test('basic agent usage, fixed ip', function (t) {
 			agent.createPool('127.0.0.1', {});
 		});
 
+		t.notOk(agent.isStopped());
+
 		agent.stop(function () {
 			t.ok(pool.isInState('stopped'));
+			t.ok(agent.isStopped());
 			t.end();
 		});
 	});
-- 
2.21.0

